client/src/lib/schemas.ts

```diff
@@ -7,24 +7,35 @@ const AddressSchema = yup.object().shape({
 });
 
 const GasPriceSchema = yup.object().shape({
-  gasPrice: yup
-    .number()
-    .required()
-    .positive(),
+  gasPrice: yup.number().required().positive(),
+});
+
+const EMAIL_REGEX = /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/;
+const ETH_REGEX = /.*\.eth$/;
+
+const AddressPageSchema = yup.object().shape({
+  address: yup
+    .mixed()
+    .test({
+      test: (value) => {
+        if (EMAIL_REGEX.test(value) || ETH_REGEX.test(value)) {
+          return true;
+        }
+
+        return false;
+      },
+    })
+    .required(),
```

Comment:
Would prefer an NPM package since this regex is not tested and nobody is going to maintain it

https://www.npmjs.com/package/email-regex

```suggestion
import emailRegex from 'email-regex';

const ETH_REGEX = /.*\.eth$/;
const isValidEmailOrEth = (value) => emailRegex({ exact: true }).test(value) || ETH_REGEX.test(value));

const AddressPageSchema = yup.object().shape({
  address: yup
    .mixed()
    .test({ test: isValidEmailOrEth })
    .required(),
```